internal/middleware/authn/authn.go

```diff
@@ -1,4 +1,4 @@
-package middleware
+package authn
```

Comment:
Moved this back into `internal` as we don't necessarily need to expose it as part of the public API yet. If we find we need it externally or get interest from the community in some way, then we can make it public. But until then let's keep things private.

Reply:
please don't move around files in the same PR that you are changing them as it makes it hard to see what actually changed